api/cluster.go

```diff
@@ -413,6 +414,87 @@ func GetClusterDetails(c *gin.Context) {
 	c.JSON(http.StatusOK, details)
 }
 
+// GetNodePools fetch node pool info for a cluster
+func GetNodePools(c *gin.Context) {
+	commonCluster, ok := getClusterFromRequest(c)
+	if ok != true {
+		return
+	}
+
+	clusterStatus, err := commonCluster.GetStatus()
+	if err != nil {
+		log.Errorf("Error getting node pool info: %s", err.Error())
+		c.JSON(http.StatusBadRequest, pkgCommon.ErrorResponse{
+			Code:    http.StatusBadRequest,
+			Message: "Error getting cluster",
+			Error:   err.Error(),
+		})
+		return
+	}
+
+	clusterTotalResources := make(map[string]float64)
+	nodePoolCounts := getActualNodeCounts(commonCluster)
+	nodePoolStatus := make(map[string]*pkgCluster.ActualNodePoolStatus)
+
+	headNodePoolName := viper.GetString(config.PipelineHeadNodePoolName)
+	for nodePoolName, nodePool := range clusterStatus.NodePools {
+		nodePoolStatus[nodePoolName] = &pkgCluster.ActualNodePoolStatus{
+			NodePoolStatus: *nodePool,
+			ActualCount:    nodePoolCounts[nodePoolName],
+		}
+
+		if nodePoolName == headNodePoolName {
+			continue
+		}
+		machineDetails, err := cloudinfo.GetMachineDetails(clusterStatus.Cloud,
+			clusterStatus.Distribution,
+			clusterStatus.Region,
+			nodePool.InstanceType)
+		if err != nil {
+			log.Warn(err.Error())
+		} else if machineDetails != nil {
+			clusterTotalResources["cpu"] += float64(nodePool.Count) * machineDetails.Cpus
+			clusterTotalResources["gpu"] += float64(nodePool.Count) * machineDetails.Gpus
+			clusterTotalResources["mem"] += float64(nodePool.Count) * machineDetails.Mem
+		}
+	}
+
+	response := pkgCluster.GetNodePoolsResponse{
+		NodePools:             nodePoolStatus,
+		ClusterTotalResources: clusterTotalResources,
+		ClusterStatus:         clusterStatus.Status,
+	}
+
+	c.JSON(http.StatusOK, response)
+}
+
+func getActualNodeCounts(commonCluster cluster.CommonCluster) map[string]int {
+	nodePoolCounts := make(map[string]int)
+	kubeConfig, err := commonCluster.GetK8sConfig()
+	if err != nil {
+		return nodePoolCounts
```

Comment:
Is it fine to return and empty collection instead of returning the error? If yes than we should log it with the appropriate context.